### ISSUE TITLE
Alt text: SVG icon representing site health status

### ISSUE THREAD

**Issue by joedolson** [WordPress/health-check#276]
### Feature request/bug description

Percentage indicator representing site health status has no screen-reader accessible text.  See: https://css-tricks.com/accessible-svgs/

**Comment by Clorith** [comment]
I ended up at [the MDN progressbar ARIA documentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role) when researching this just now, and I wonder if it would be acceptable to use a combination of tools here.

The `aria-valuenow` gets set when we know the percentage and the tests are done, but we still maintain the `wp.a11y.speak` to give a friendly indicator that the testing is done, this way they know the context of the percentage change.

At the same time, introducing `aria-busy` on the content area following the header (on the `health-check-body` class) until the testing is done. This would hopefully address the concern I had with some of the tests running async, but I would like your input on the matter in case I misunderstood it.

**Comment by joedolson** [comment]
These interactions are complex enough that I'd really need a live version to test. Outside of testing, the last I know about aria-busy is that it is only supported by JAWS, and not NVDA or VoiceOver, which would significantly limit the value of that. 

aria-hidden can serve the same purpose, and has better support. See: https://developer.paciellogroup.com/blog/2018/05/short-note-on-being-busy/

**Comment by Clorith** [comment]
I'd be happy to build you a fresh copy of it with the changes I made with `aria-busy` and `valuenow` (I'll replace `busy` with `hidden` though).

**Comment by joedolson** [comment]
That would be helpful!

**Comment by Clorith** [comment]
Excellent, here's a new copy, I must admit I don't recall what other improvements had made it into this branch at the time, so some things that have been resolved may not be included in this one.

[health-check.zip](https://github.com/WordPress/health-check/files/2916791/health-check.zip)

**Comment by joedolson** [comment]
Using NVDA with Firefox, I can't locate the site health percentage, and I don't get any kind of announcement about it. 

I think the main problem here is that you're not providing any kind of textvalue for the progressbar; there's no aria-valuenow (or aria-valuetext), no element text node, etc. There's nothing in the progressbar for a screen reader to interpret.

I'm not sure of the support of aria-valuetext and progressbar elements, but with *no* elements in this context that communicate information to a screen-reader, your primary problem is that there's nothing to read - whether it gets announced at the right time is a secondary concern, at this point. 

Look at the article above on SVG accessibility to learn more about exposing the meaningful text in SVG, and try adding aria-valuetext or aria-valuenow attributes.

**Comment by Clorith** [comment]
Hmm, that's concerning then, I checked and it should have a `aria-valuenow` (and min/max equivalents), and the value does get updated when the tests are done running.

The documentation said the `aria-valuenow` will be announced as a percentage calculated off the `aria-valuemax`, I'll get NVDA running so I can do some local testing as well, as this is a valuable indicator for users, so it should be available to all, or none. (It is possible to remove that feature for a core inclusion, and work on it as a future plugin feature, but I'd obviously want to get this included, so I'll double down on it :+1: )

**Comment by joedolson** [comment]
Hmm. aria-valuenow doesn't seem to be in my current copy...is the version number meaningful? What I've got is labeled 1.3.0-beta1, but I'm guessing the on-the-fly version you generated may not have a unique version. 

Just re-installing now from the above link.

**Comment by joedolson** [comment]
Nope - not there.

**Comment by Clorith** [comment]
Interesting, I've tested with NVDA now, I can see the code, but it's not being picked up, so I'll do some testing with this until I figure out why it's not being picked up, I suspect it's a matter of object hierarchy so we'll get that sorted

**Comment by Clorith** [comment]
After a bit of testing, I've come to the conclusion that `aria-valuetext` is required regardless, even though documentation said it wasn't.

I've implemented the `aria-valuenow`, I wanted to provided an initial string of **Site tests are running, please wait a moment.**, to give feedback on what is happening, unfortunately it's not possible to update the `valuenow` text, it only reads the initial string on page load if one is set, so to announce tests are completed and their pass percentage this needs to be left blank.

Once tests are completed we update the string, and also expose the rest of the page to the screenreader by removing the `aria-hidden` as discussed. I went with the string **All site health tests have finished running. Your site scored %s.** for the end result.

I've attached an updated build with these modifications (I've bumped versions to make it easier to pick up changes, it should be `1.3.0-beta-2019-03-04-16-36`), as my understanding (or use) of NVDA may not be expected so interactions may differ.

[health-check.zip](https://github.com/WordPress/health-check/files/2927309/health-check.zip)

**Comment by Clorith** [comment]
I just can't make up my mind now, can I.

I've been playing more with it since yesterday, and the one I landed on now makes me quite happy. It uses `aria-valuenow` to indicate tests are still running, by pre-filling the field. It then uses `wp.a11y.speak` when the tests are done to inform you that tests are done, and results are ready.

The new completion string is `All site health tests have finished running. Your site scored %s, and the results are now available on the page.`. I chose to include the last bit about results being available now, as the screen reader may already have read the body content, skipping the results section (as it's currently marked with `aria-hidden` while tests run).

I've made a second build, with a new version string (`1.3.0-beta-2019-03-05-10-04`) for your convenience here, which uses this approach instead.
[health-check.zip](https://github.com/WordPress/health-check/files/2930096/health-check.zip)

This is my preferred one I think, it feels much smoother from a reader-perspective, as it allows us to provide context at multiple stages, instead of just at the end which we were limited to when only using `aria-valuenow`.

**Comment by Clorith** [comment]
Implemented this in #302, which appears to work well. If the implementation should be incorrectly treating some scenarios I would love for them to go into separate tickets just so they don't get missed 👍

**Comment by afercia** [comment]
> Using NVDA with Firefox, I can't locate the site health percentage, and I don't get any kind of announcement about it.

For history, this has been addressed in core in https://core.trac.wordpress.org/ticket/46621 by removing any ARIA related to progressbar and by using some screen-reader-text.
